docker/main/ngen/funcs.sh

```diff
@@ -92,15 +92,24 @@ exec_main_worker_ngen_run()
 
     # Execute the model on the linked data
     echo "$(print_date) Executing mpirun command for ngen on ${MPI_NODE_COUNT} workers"
-    ${MPI_RUN:?} -f "${MPI_HOSTS_FILE}" -n ${MPI_NODE_COUNT} \
-        ${NGEN_EXECUTABLE:?} ${HYDROFABRIC_DATASET_DIR}/catchment_data.geojson "" \
-                ${HYDROFABRIC_DATASET_DIR}/nexus_data.geojson "" \
-                ${CONFIG_DATASET_DIR}/realization_config.json \
-                ${PARTITION_DATASET_DIR}/partition_config.json \
-                --subdivided-hydrofabric
-
-    #Capture the return value to use as service exit code
-    NGEN_RETURN=$?
+    if [ -e ${HYDROFABRIC_DATASET_DIR}/catchment_data.geojson ]; then
```

Comment:
These are probably more evidence that we need to bump the priority for the ticket for moving logic into python. It might just be because I generally avoid too much shell scripting, but I find this hard to read/track. It doesn't break convention with the rest of the script, so moving that out to another language isn't in the scope of this task.